firedrake/mesh.py

```diff
@@ -1085,6 +1094,15 @@ def coordinates(self, value):
 
         raise AttributeError(message)
 
+    def update_aux_coordinates(self):
+        """
+        Updates all cached aux-coordinate functions.
+
+        Mesh coordinates are re-interpolated on aux functions.
+        """
+        for _, interpolator in self.aux_coordinate_functions.values():
+            interpolator.interpolate()
```

Comment:
It's also unclear to me what is the best API to invalidate the cached coordinates. One option would be to store a dirty bool flag for each aux coordinate, which would trigger interpolation next time the function is accessed.